### PR TITLE
Service Registry API

### PR DESCRIPTION
**Pull request by IstrateM**
Created service registry API

### CODE REVIEW

File: api/serviceregistry/serviceregistry.proto [hunk 1]
```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+
+package svreg;
+option go_package = "serviceregistry";
+
+
+service Registry{
+  rpc RegisterWorkload (ServiceWorkload) returns (Empty) {} // Request register of a new service
+  rpc QueryService (ServiceRequest) returns (ServiceWorkloads) {} // Request a services
```
Comment (adodon2go) [inline api/serviceregistry/serviceregistry.proto]: Comment should be updated to something similar to "Request workloads for a particular service"

File: api/serviceregistry/serviceregistry.proto [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+//Basic workload
+message Workload {
+  WorkloadIdentifier identifier = 1;
+  repeated string IPAddress = 2;
+  string connectivityDomain = 3;
+}
+
+//Request used to querry
```
Comment (adodon2go) [inline api/serviceregistry/serviceregistry.proto]: typo

**Review by adodon2go — APPROVED** [review]

File: api/serviceregistry/helper.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package serviceregistry
+
+type validationErrors []error
```
Comment (mihmatache) [inline api/serviceregistry/helper.go:9]: This helper file seems to be useless right now. I think adding some validation checks here could help.

File: api/serviceregistry/serviceregistry.proto [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+service Registry{
+  rpc RegisterWorkload (ServiceWorkload) returns (Empty) {} // Request register of a new service
+  rpc QueryService (ServiceRequest) returns (ServiceWorkloads) {} // Request a services
+  rpc RemoveWorkload (ServiceWorkload) returns (Empty) {} // Remove service from list
+}
+
+message Empty {}
+
+//The request used to register a new service or update the existing ones
+message ServiceWorkload {
```
Comment (mihmatache) [inline api/serviceregistry/serviceregistry.proto:23]: Do we really need both a singular and a plural for this? I think the plural can be sufficient. Either that, or return a list of singulars. there is no point in having two API objects with 90% the same code.

File: api/serviceregistry/serviceregistry.proto [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  string connectivityDomain = 2;
+  Workload workload = 3;
+  repeated int32 ports = 4;
+}
+
+//Basic workload
+message Workload {
+  WorkloadIdentifier identifier = 1;
+  repeated string IPAddress = 2;
+  string connectivityDomain = 3;
```
Comment (mihmatache) [inline api/serviceregistry/serviceregistry.proto]: The connectivity domain information is present in both the Workload objects and the Service objects. Do we need it on both places?

File: api/serviceregistry/serviceregistry.proto [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+//The request used to register a new service or update the existing ones
+message ServiceWorkload {
+  string serviceName = 1;
+  string connectivityDomain = 2;
+  Workload workload = 3;
+  repeated int32 ports = 4;
+}
+
+//Basic workload
```
Comment (mihmatache) [inline api/serviceregistry/serviceregistry.proto:30]: Style nit: Please try to keep items that work together next to each other in the proto definition. The Workload item is here, in between 2 Service type items, and the WorkloadIdentifier, which is only used by the workload is at the bottom.

**Review by mihmatache — CHANGES_REQUESTED** [review]
Please cut down a bit on the number of slightly different API response/request types. This will cause confusion when using them.

**Review by ana-moga — APPROVED** [review]

File: api/serviceregistry/serviceregistry.proto (hunk 4 above)
Comment (IstrateM) [reply api/serviceregistry/serviceregistry.proto:23]: done

File: api/serviceregistry/serviceregistry.proto (hunk 5 above)
Comment (IstrateM) [reply api/serviceregistry/serviceregistry.proto]: done

File: api/serviceregistry/serviceregistry.proto (hunk 6 above)
Comment (IstrateM) [reply api/serviceregistry/serviceregistry.proto:30]: done

File: api/serviceregistry/serviceregistry.proto (hunk 2 above)
Comment (IstrateM) [reply api/serviceregistry/serviceregistry.proto]: solved

File: api/serviceregistry/serviceregistry.proto (hunk 1 above)
Comment (IstrateM) [reply api/serviceregistry/serviceregistry.proto]: done

**Review by mihmatache — APPROVED** [review]

**Pull request merged**